pyface/ui/__init__.py

```diff
@@ -0,0 +1,84 @@
+# (C) Copyright 2005-2023 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+from importlib import import_module
+from importlib.abc import MetaPathFinder, Loader
+from importlib.util import find_spec
+from importlib.machinery import ModuleSpec
+import sys
+
+
+# Import hooks for loading pyface.ui.qt.* in place of a pyface.ui.qt4.*
+# This is just the implementation, it is not connected in this module, but
+# is available for applications which want to install it themselves.
+# It is here rather than in pyface.ui.qt4 so it can be imported and used
+# without generating the warnings from pyface.ui.qt4
+#
+# To use manually:
+#
+#     import sys
+#     sys.meta_path.append(ShadowedModuleFinder())
+
+class ShadowedModuleLoader(Loader):
+    """This loads another module into sys.modules with the given name.
+    """
+
+    def __init__(self, fullname, new_name, new_spec):
+        self.fullname = fullname
+        self.new_name = new_name
+        self.new_spec = new_spec
+
+    def create_module(self, spec):
+        if self.new_name not in sys.modules:
+            import_module(self.new_name)
```

Comment:
I'm not sure that this is right - from my reading of the docs, it should be `exec_module` that's doing the work of actually populating the module, while `create_module` should just be providing the unpopulated module object. And I believe neither `create_module` nor `exec_module` should be causing changes to `sys.modules` - I think the import machinery takes care of that.

I'm not sure under what circumstances this would matter. I guess we lose the ability to `reload` (since that would presumably be re-executing `exec_module` but not `create_module`), but that doesn't seem like a big loss. Having `create_module` potentially alter `sys.modules` worries me a bit more - I'm not sure what the ramifications of that could be.